http-client/src/main/java/io/airlift/http/client/Request.java

```diff
@@ -221,6 +246,8 @@ public static Builder fromRequest(Request request)
         private Optional<HttpVersion> version = Optional.empty();
         private boolean followRedirects = true;
         private boolean preserveAuthorizationOnRedirect;
+        private Duration requestTimeout = Duration.ZERO;
```

Comment:
It seems better to make these `Optional`. We actually allow zero for both of these in `HttpClientConfig` (though I'm not sure what that means in the context of a request timeout).